fem/fespace.cpp

```diff
@@ -1193,22 +1231,42 @@ void FiniteElementSpace::Construct()
 
    if (mesh->Dimension() == 3 && mesh->GetNE())
    {
-      // Here we assume that all faces in the mesh have the same base
-      // geometry -- the base geometry of the 0-th face element.
-      // The class Mesh assumes the same inside GetFaceBaseGeometry(...).
-      // Thus we do not need to generate all the faces in the mesh
-      // if we do not need them.
-      int fdof = fec->DofForGeometry(mesh->GetFaceBaseGeometry(0));
-      if (fdof > 0)
+      Geometry::Type face_geom = mesh->GetFaceBaseGeometry(-1);
+      MFEM_VERIFY(face_geom != Geometry::INVALID,
```

Comment:
I always like to have a value like this in enumerations.  I use it to initialize variables of the enumerated type and to check for errors.  For example an empty mesh is created with base geometry equal to Geometry::INVALID because no other member of that enumeration would make sense.